tf2/src/buffer_core.cpp

```diff
@@ -834,18 +836,23 @@ bool BufferCore::canTransform(const std::string& target_frame, const std::string
                            const TimePoint& time, std::string* error_msg) const
 {
   // Short circuit if target_frame == source_frame
-  if (target_frame == source_frame)
-    return true;
-
-  if (warnFrameId("canTransform argument target_frame", target_frame))
-    return false;
-  if (warnFrameId("canTransform argument source_frame", source_frame))
-    return false;
+  if (target_frame == source_frame) return true;
 
   std::unique_lock<std::mutex> lock(frame_mutex_);
 
-  CompactFrameID target_id = lookupFrameNumber(target_frame);
-  CompactFrameID source_id = lookupFrameNumber(source_frame);
+  CompactFrameID target_id, source_id;
+
+  try {
```

Comment:
That sounds like a great idea for a followup PR by someone who is more comfortable than I with this code

Reply:
This logic is creating the warnings, throwing an exception then immediately catching the exception and then pulling the string out of it. This is in one of the potentially highest throughput and consequently desired to be lowest latency sections of code. We don't want to be doing that. 

If we add the third check of non-existant frame to warnFrameId it will have full coverage of the cases for `validateFrameId`'s cases. And right now `warnFrameId` only outputs to console. In some cases like `can_transform` where we have an error string that we can populate if it's set. We should add that same capability to `warnFrameId` by adding the `error_msg` argument and then it can simply be added as an additional argument.  And the behavior can be to either populate that `error_msg` if it's present, otherwise do the current behavior and send a warning to the console.

This will be an overloaded function and the current prototype can be updated to simply call the new version with the `error_msg` set to null to get the same behavior.